peer/network.go

```diff
@@ -505,12 +505,11 @@ func (n *network) Connected(ctx context.Context, nodeID ids.NodeID, nodeVersion
 		return nil
 	}
 
-	if nodeID == n.self {
-		log.Debug("skipping registering self as peer")
-		return nil
+	if nodeID != n.self {
+		// The legacy peer tracker doesn't expect to be connected to itself.
+		n.peers.Connected(nodeID, nodeVersion)
 	}
```

Comment:
I guess it's safe but should we also skip disconnecting self in legacy peer tracker as well? (or at least we can add a comment)